.circleci/config.yml

```diff
@@ -0,0 +1,67 @@
+version: 2
+workflows:
+  version: 2
+  test-and-build:
+    jobs:
+      - test
+      - build:
+          requires:
+            - test
+jobs:
+  build:
+    working_directory: ~/prison-visits-integration-tests
+    docker:
+      - image: docker:17.09.0-ce-git
+    steps:
+      - checkout
+      - setup_remote_docker
+      - type: shell
+        name: Push application Docker image
+        command: |
+          docker login -u $DOCKER_USER -p $DOCKER_PASS
+          docker build -t "stupidcodefactory/prison-visits-integration-tests:${CIRCLE_BRANCH}" .
+          docker push "stupidcodefactory/prison-visits-integration-tests:${CIRCLE_BRANCH}"
+
+
+  test:
+    parallelism: 1
+    working_directory: ~/prison-visits-integration-tests
+    docker:
+      - image: circleci/ruby:latest-browsers
+    steps:
+      - checkout
+
+      # Restore bundle cache
+      - restore_cache:
+          key: rails-demo-{{ checksum "Gemfile.lock" }}
```

Comment:
The keys don't match, shouldn't it be `prison-visits-2-{{ checksum "Gemfile.lock" }}`